internal/rbd/nodeserver.go

```diff
@@ -189,33 +186,13 @@ func populateRbdVol(
 		}
 		rv.RbdImageName = volID
 	} else {
-		var vi util.CSIIdentifier
-		var imageAttributes *journal.ImageAttributes
-		err = vi.DecomposeCSIID(volID)
+		rv, err = GenVolFromVolID(ctx, volID, cr, req.GetSecrets())
```

Comment:
`rv` was just created at line 174. Why is it needed to create it again (and potentially leak something)?

Reply:
We need to reassign to `rv` after finding the right rbdVolume from the clusterID mapping. More details are provided in the issue https://github.com/ceph/ceph-csi/issues/2929#issue-1165094633

Reply:
Let me see if i can rearrange the code block, its been many months.

Reply:
That is not clear from the code. Please add comments that explain this. Or, assign `rv` only once, move line 174 inside the `if isStaticVol`?

Reply:
Code is rearranged now, PTAL

Reply:
thanks!